### PR TITLE
cleaning up <Loading /> implementation to prevent jumpy pageload

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: extension/src/popup/App.tsx [hunk 1]
```diff
@@ -59,7 +66,10 @@ export function App() {
   return (
     <Provider store={store}>
       <GlobalStyle />
-      <Router />
+      <RouteWrapperEl>
+        <Router />
+      </RouteWrapperEl>
+      <Loading />
```
Comment (piyalbasu) [inline extension/src/popup/App.tsx:72]: keep the <Loading /> component behind the Router component so if the Route hasn't resolved, the loading screen will show

File: extension/src/popup/Router.tsx [hunk 2]
```diff
@@ -55,9 +53,11 @@ const PublicKeyRoute = (props: RouteProps) => {
   if (applicationState === APPLICATION_STATE.APPLICATION_ERROR) {
     return <AppError>{error}</AppError>;
   }
+
   if (applicationState === APPLICATION_STATE.APPLICATION_LOADING) {
-    return <Loading />;
+    return null;
```
Comment (piyalbasu) [inline extension/src/popup/Router.tsx:58]: returning `null` will show the <Loading /> component

File: extension/src/popup/Router.tsx [hunk 3]
```diff
@@ -193,25 +194,21 @@ export const Router = () => {
       <RouteListener />
       <Switch>
         <PublicKeyRoute path={ROUTES.account}>
-          <Header />
```
Comment (piyalbasu) [inline extension/src/popup/Router.tsx:196]: It's a better implementation to just include <Header /> inside views that need it.

File: extension/src/popup/components/Header.tsx [hunk 4]
```diff
@@ -42,12 +38,6 @@ type HeaderProps = {
 };
 
 export const Header = ({ className, ...props }: HeaderProps) => {
-  const applicationState = useSelector(applicationStateSelector);
-
-  if (applicationState === APPLICATION_STATE.APPLICATION_LOADING) {
-    return null;
-  }
```
Comment (piyalbasu) [inline extension/src/popup/components/Header.tsx:49]: this was a dumb idea i tried near the beginning and was preventing views that required the Header from just including <Header /> in their render

File: extension/src/popup/views/Account.tsx [hunk 5]
```diff
@@ -164,5 +166,5 @@ export const Account = () => {
       </AccountEl>
       <Footer />
     </>
-  );
+  ) : null;
```
Comment (piyalbasu) [inline extension/src/popup/views/Account.tsx:169]: Now we'll just be seeing the Loading screen until this component returns something

File: extension/src/popup/views/AccountCreator.tsx [hunk 6]
```diff
@@ -65,62 +66,65 @@ export const AccountCreator = () => {
...
-      header="Create a password"
-      icon={CreatePasswordIllo}
-      goBack={() => navigateTo(ROUTES.welcome)}
-    >
-      <Formik
-        initialValues={initialValues}
-        onSubmit={handleSubmit}
-        validationSchema={AccountCreatorSchema}
+    <>
+      <Header />
```
Comment (piyalbasu) [inline extension/src/popup/views/AccountCreator.tsx:70]: much nicer to include Header here

File: extension/src/popup/views/Loading.tsx [hunk 7]
```diff
@@ -8,8 +18,39 @@ const LoadingEl = styled.div`
...
+  }
+
+  @keyframes lds-dual-ring {
+    0% {
+      transform: rotate(0deg);
+    }
+    100% {
+      transform: rotate(360deg);
+    }
+  }
```
Comment (piyalbasu) [inline extension/src/popup/views/Loading.tsx:46]: this is an open source css loading spinner. This is placeholder until design comes up with something custom to implement

File: extension/src/popup/App.tsx (hunk 1 above)
Comment (vcarl) [reply extension/src/popup/App.tsx:72]: Ah interesting, so this is like a backdrop that gets covered up if there isn't something else to show?

File: extension/src/popup/views/Loading.tsx (hunk 7 above)
Comment (vcarl) [reply extension/src/popup/views/Loading.tsx:46]: Maybe nice to include a link to the source of it, in that case

**Review by vcarl — APPROVED** [review]

File: extension/src/popup/App.tsx (hunk 1 above)
Comment (piyalbasu) [reply extension/src/popup/App.tsx:72]: yup, that's right

**Pull request merged**